esp/esp/users/models/__init__.py

```diff
@@ -1038,6 +1038,13 @@ def getRankInClass(student, subject, default=10):
             rank = default
         return rank
 
+@dispatch.receiver(signals.post_save, sender=User,
+                     dispatch_uid='make_admin_save')
+def make_admin_save(sender, instance, **kwargs):
+    if instance.is_superuser:
+        from esp.users.views.make_admin import make_user_admin
+        make_user_admin(ESPUser(instance))
```

Comment:
I like this approach, but this way of writing it won't work once #1862 is merged.